src/Authentication.Abstractions/Interfaces/IAuthenticationFactory.cs

```diff
@@ -94,5 +95,10 @@ public interface IAuthenticationFactory : IHyakAuthenticationFactory
         /// <param name="account">The account to remove credentials for</param>
         /// <param name="tokenCache">The TokenCache to remove credentials from</param>
         void RemoveUser(IAzureAccount account, IAzureTokenCache tokenCache);
+
+        /// <summary>
+        /// Get the information to be recorded in Telemetry
+        /// </summary>
+        IList<IAuthenticationInfo> GetDataForTelemetry();
```

Comment:
Updating a core interface can be risky. We must

1. Update all implementation of this interface. I believe there's one in Accounts and another in Accounts' tests.
2. Test collaboration of new Accounts and older version of service modules, especially if they use this interface.